src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -2711,9 +2725,10 @@ class VariantSampleMatrix(val hc: HailContext, val metadata: VSMMetadata,
   }
   
   def toIndexedRowMatrix(expr: String, getVariants: Boolean): (Option[Array[Any]], IndexedRowMatrix) = {
-    val partitionSizes = rdd2.mapPartitions(it => Iterator.single(it.size)).collect().scanLeft(0)(_ + _)
-    assert(partitionSizes.length == rdd2.getNumPartitions + 1)
-    val pSizeBc = sparkContext.broadcast(partitionSizes)
+    val pCounts = partitionCounts()
+    val partitionStarts = pCounts.scanLeft(0L)(_ + _)
```

Comment:
I think it's worth moving the partitionStarts def to VSM under partitionCounts, it's already in two places and will be in more.